server/src/core/event/event.resolver.ts

```diff
@@ -0,0 +1,57 @@
+import { Resolver, Mutation, Args, Context } from '@nestjs/graphql';
+import { EventService } from './event.service';
+import { Event } from './event.entity';
+import { CreateEventInput } from './dto/create-event.input';
+import axios from 'axios';
+import { OptionalJwtAuthGuard } from 'src/guards/optional-jwt.auth.guard';
+import { UseGuards } from '@nestjs/common';
+import { AuthWorkspace } from 'src/decorators/auth-workspace.decorator';
+import { User, Workspace } from '@prisma/client';
+import { AuthUser } from 'src/decorators/auth-user.decorator';
+import crypto from 'crypto';
+
+@UseGuards(OptionalJwtAuthGuard)
+@Resolver(() => Event)
+export class EventResolver {
+  constructor(private readonly eventService: EventService) {}
+
+  @Mutation(() => Event)
+  createEvent(
+    @Args() createEventInput: CreateEventInput,
+    @Context() context: any,
+    @AuthWorkspace() workspace: Workspace,
+    @AuthUser() user: User,
+  ) {
+    if (process.env.TELEMETRY_ENABLED === 'false') {
+      return;
+    }
+
+    function anonymize(input) {
+      if (process.env.IS_TWENTY_CLOUD === 'true') {
```

Comment:
we should avoid this type of check: IS_ENV / IS_TWENTY_CLOUD as they do not scale well. Instead, let's use a more "configurable" pattern: IS_TELEMETRY_ENABLE / IS_TELEMETRY_ANONYMIZATION_ENABLE (we will set it to false on cloud for example)

Reply:
OK!